dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java

```diff
@@ -698,6 +699,43 @@ public Map<String, Object> verifyProcessDefinitionName(User loginUser, long proj
         return result;
     }
 
+    @Override
```

Comment:
We need to add `@Transaction` to rollback the data.

Reply:
Batch deletion, implemented as a loop call `deleteProcessDefinitionByCode` which add `@Transaction`. In my original idea, the delete operation of each workflow does not affect each other.
So, I would like to ask, it is necessary to add `@Transaction` here?

Reply:
If the operation failed, but some workflow has been deleted, this will make the data incorrect.